src/XMakeTasks/CodeTaskFactory.cs

```diff
@@ -771,6 +771,16 @@ private Assembly CompileInMemoryAssembly()
                 if (!s_compiledTaskCache.TryGetValue(fullSpec, out existingAssembly))
                 {
                     // Invokes compilation. 
+
+                    // Note: CompileAssemblyFromSource uses Path.GetTempPath() directory, but will not create it. In some cases 
+                    // this will throw inside CompileAssemblyFromSource. To work around this, ensure the temp directory exists. 
+                    // See: https://github.com/Microsoft/msbuild/issues/328
+                    var tempPath = Path.GetTempPath();
+                    if (!Directory.Exists(tempPath))
```

Comment:
I did that first, but read through the code for [CreateDirectory](http://referencesource.microsoft.com/mscorlib/a.html#5c99436f88797b6a) and it looked like some small differences in permissions it checked before it checks for Exists. The last thing I wanted was to throw here for something like that.

Looks like there's a stackoverflow question on [this](http://stackoverflow.com/questions/27575384/c-sharp-directory-createdirectory-path-should-i-check-if-path-exists-first) and just CreateDirectory appears to be the winner. But not without some dissent :). I imagine just CreateDirectory is fine, just overly paranoid.